File: purchase-insurance-service/src/main/java/com/stackroute/purchaseinsuranceservice/model/CustomerInsurancePurchase.java
package com.stackroute.purchaseinsuranceservice.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.mapping.Document;

@Document
@Data
@NoArgsConstructor
@AllArgsConstructor
public class CustomerInsurancePurchase {
    private String customerPolicyId;
    private String insurancePolicyId;

    private String email;

    private long sumInsured;
    private String startDate;
    private String purchaseDate;
    private String endDate;
    private int duration;
    private String[] addOnName;
    private int premium;
    private String name;
    private long mobile;
    private String address;
    private int pincode;
    private String city;
    private String state;
    private String nameOfNominee;
    private String nomineeDOB;
    private String relation;
    private HealthInsurance healthInsurance;
    private LifeInsurance lifeInsurance;
    private AutomobileInsurance automobileInsurance;

    private boolean renewalStatus=false;
    private boolean status=true;

}

File: product-webapp/app/src/app/CustomerInsurancePurchase.ts
import { AutomobileInsurance } from "./AutomobileInsurance";
import { HealthInsurance } from "./HealthInsurance";
import { LifeInsurance } from "./LifeInsurance";

export class CustomerInsurancePurchase{
    constructor(
       public customerPolicyId :string,
       public insurancePolicyId :string,
       public email :string,
       public sumInsured :number,
       public startDate :string,
       public endDate :string,
       public duration :number,
       public addOnName :string[],
       public premium :number,
       public name :string,
       public mobile :number,
       public address :string,
       public pincode :number,
       public city :string,
       public state :string,
       public nameOfNominee :string,
       public nomineeDOB :string,
       public relation :string,
       public renewalStatus :boolean=false,
       public status :boolean=true,
       public healthInsurance:HealthInsurance,
       public lifeInsurance:LifeInsurance,
       public automobileInsurance:AutomobileInsurance
    ){ }
}